### PR TITLE
Finalized doxygen pages

### PR DESCRIPTION
**Pull request by mmattar**
- Limited to C# classes that are user-facing
- Added a placeholder for Python to signify intentions
- Generated doxygen pages

### CODE REVIEW

**Comment by vincentpierre** [comment]
Are we supposed to have the doxygen documents in the repo ? Shouldn't this be hosted somewhere ?

**Comment by vladimiroster** [comment]
Keeping generated files in the repo is usually bad practice.

**Comment by jo3w4rd** [comment]
In this case, it lets us host the docs, including the API reference, from Github (using github pages).

**Comment by mmattar** [comment]
@jo3w4rd - we should do that for the next (minor) release. For now, I've updated the pages to provide documentation on how to generate them yourself.

@vincentpierre + @vladimiroster : please review new version.

**Comment by xiaomaogy** [comment]
Do we have a ml-agents.conf file anywhere? I don't see it in the repo, and not in this pull request, but it is noted in the API-Reference.md.

**Comment by jo3w4rd** [comment]
The file is named "dox-ml-agents.conf". Were you planning to rename it @mmattar?

**Comment by mmattar** [comment]
Great catch folks - the current name works for me. Just pushed a change.

File: docs/API-Reference.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+[Doxygen](http://www.stack.nl/~dimitri/doxygen/) for auto-generating HTML 
+documentation.
+
+To generate the API reference, 
+[download Doxygen](http://www.stack.nl/~dimitri/doxygen/download.html) and run
+the following command within the `docs/` directory:
+
+    doxygen dox-ml-agents.conf
+
+`ml-agents.conf` is a configuration file that includes the classes
```
Comment (xiaomaogy) [inline docs/API-Reference.md]: ml-agents.conf ---> dox-ml-agents.conf

File: docs/dox-ml-agents.conf [hunk 2]
```diff
@@ -783,33 +783,10 @@ WARN_LOGFILE           =
...
-                         Training-Imitation-Learning.md \
-                         Using-TensorFlow-Sharp-in-Unity.md \
-                         Training-on-Amazon-Web-Service.md \
-                         Using-Tensorboard.md \
-                         Python-API.md \
-                         Feature-Broadcasting.md \
-                         Feature-Monitor.md \
-                         Limitations-and-Common-Issues.md \
-                         ../unity-environment/Assets/ML-Agents/Scripts \
-                         ../python/unityagents
```
Comment (xiaomaogy) [inline docs/dox-ml-agents.conf]: Why do we only include a portion of the files in the ../unity-environment/Assets/ML-Agents/Scripts and ignoring files like Communicator.cs and BCTeacherHelper.cs?

Comment (mmattar) [reply docs/dox-ml-agents.conf]: We've only included the files that the developers will be using or extending. We do need to add ALL in the future.

File: docs/API-Reference.md (hunk 1 above)
Comment (mmattar) [reply docs/API-Reference.md]: Great catch - fixed!

**Pull request merged**